logging/options.go

```diff
@@ -0,0 +1,43 @@
+package logging
+
+import (
+	"io"
+)
+
+type Option func(s *settings)
+
+func SetLevel(level Level) Option {
+	return func(s *settings) {
```

Comment:
closures are cool but might be overkill here? idk

Reply:
I agree I don't like complicated suff. The advantage of this 'options pattern' is that the user can use

```go
logging.New(SetLevel(LevelInfo), SetSomethingElse("blabla"))
```

with variadic options, or not pass any option at all. It's just more convenient/shorter end user wise than passing a `struct` with a bunch of fields you are not sure you need to set or not.

Reply:
agreed 👍